matrix/src/dense.rs

```diff
@@ -34,8 +34,14 @@ impl<T, S: Borrow<[T]> + Into<Vec<T>> + Send + Sync> DenseStorage<T> for S {}
 impl<T: Clone + Send + Sync, S: DenseStorage<T>> DenseMatrix<T, S> {
     #[must_use]
     pub fn new(values: S, width: usize) -> Self {
-        debug_assert!(width >= 1);
-        debug_assert_eq!(values.borrow().len() % width, 0);
+        debug_assert_eq!(
```

Comment:
nit: reads a bit weird, maybe do `debug_assert!(width == 0 || values.borrow().len() % width == 0)`?

Or could do `if width > 0 { debug_assert_eq!(values.borrow().len() % width, 0); }`, I think it would still get removed in release mode

Reply:
Ah sorry, totally agree with this nit :)